cekit/descriptor/base.py

```diff
@@ -126,6 +126,26 @@ def process_defaults(self):
         if 'user' not in self._descriptor['run']:
             self._descriptor['run']['user'] = cekit.DEFAULT_USER
 
+    def remove_none_keys(self):
+        if isinstance(self, Descriptor):
+            _remove_none_keys(self)
+        else:
+            # it means it list
+            for desc in self:
+                _remove_none_keys(desc)
+
+
+def _remove_none_keys(desc):
+    for key in {k: v for k, v in desc.items()}:
```

Comment:
this is a shallow clone so you can delete keys, I think? this might work: `for key in dict(desc):`